drivers/timer/hpet.c

```diff
@@ -260,14 +256,20 @@ static void hpet_isr(const void *arg)
 
 	uint64_t now = hpet_counter_get();
 
-#if ((DT_INST_IRQ(0, sense) & IRQ_TYPE_LEVEL) == IRQ_TYPE_LEVEL)
 	/*
 	 * Clear interrupt only if level trigger is selected.
 	 * When edge trigger is selected, spec says only 0 can
 	 * be written.
 	 */
+#if (DT_INST_IRQ_HAS_CELL(0, sense))
+#if ((DT_INST_IRQ(0, sense) & IRQ_TYPE_LEVEL) == IRQ_TYPE_LEVEL)
+	hpet_int_sts_set(TIMER0_INT_STS);
+#endif
+#else
+#if HPET_INT_LEVEL_TRIGGER
```

Comment:
maybe all this #if/#else could be gathered into one macro function, that you could use like: 
```
#if NAME_OF_THE_MACRO()
        do_something();
#endif
```

You could reuse that below for level trigger as well. could help for readability.

Reply:
The thing is a little complicated, no easy way to gather into one macro.
In the new version I simplified it into one macro by changes from line 224 to 237. I also explained the background there.
Please review, thanks.